src/security/ddwaf_obj.cpp

```diff
@@ -0,0 +1,95 @@
+#include "ddwaf_obj.h"
+
+namespace datadog::nginx::security {
+namespace impl {
+template <typename D>
+void json_to_obj_impl(DdwafMemres &memres, ddwaf_obj &object, const D &doc) {
```

Comment:
Actually this one probably should take an optional max depth just in case we decide to use it to parse json bodies. The ruleset generation should not be limited in depth.

Reply:
This is used only to parse configuration. But I'm adding a limit anyway.